packages/block-library/src/query/edit/inspector-controls/post-format-controls.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * WordPress dependencies
+ */
+import { SelectControl } from '@wordpress/components';
+import { useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+import { decodeEntities } from '@wordpress/html-entities';
+import { __ } from '@wordpress/i18n';
+
+export default function PostFormatControls( { onChange, query } ) {
+	const { postType, postFormat } = query;
+
+	const { postFormats } = useSelect(
+		( select ) => {
+			const { getEntityRecords } = select( coreStore );
+			return {
+				postFormats: getEntityRecords( 'taxonomy', 'post_format', {
```

Comment:
So looking at the existing PostFormat control from the inspector control (post-format-controls.js). It seems like it's not using the REST API for this (or at least not this endpoint)

It's using `const supportedFormats = select( coreStore ).getThemeSupports()?.formats;`
and using that array to filter a static list of post formats defined on that file.

So my thinking is, maybe we should be consistent and do the same here as well. (We could potentially extract the list of formats to the `constants.js` file of the package and use it in both places.

I'm guessing this answers the question that you asked about the available post formats to pick from.

I'm guessing this also means that we don't really need to make the post format taxonomy's show_in_rest true anymore.

Reply:
I'm going to try it, but need to figure out how to display the correct formats per post type, not only list all the formats available in the theme.
If the theme supports all, but a **custom** post type only support one, only one should show.

Reply:
When a post is assigned a format WP saves a term taxonomy id that needs to be used in the query to get the correct content for the preview in the post template.
getThemeSupports only returns the array of names, not id's.
So if getThemeSupports is used, at some point there still needs to be a second database search that compares the name `aside` with the name or slug in wp_terms `post-format-aside` and gets the matching term id.

Reply:
> display the correct formats per post type,
It seems the block editor does not support the classic PHP filters, so I'm removing this from the todo list (for now).

Reply:
> getThemeSupports only returns the array of names, not id's

I'm I understanding correctly than when assigning a post format to a "post" object, we use the "names" and not the "ids" ? If that's the case, it seems logical that the REST API to filter posts should support a dedicated postFormat argument that accept names and not ids.

Reply:
I am trying to understand that too:

https://developer.wordpress.org/rest-api/reference/posts/#schema
https://developer.wordpress.org/rest-api/reference/posts/#schema-format
Expects a string with a name.

https://developer.wordpress.org/reference/functions/wp_set_post_terms/
Expects a string or array with either an ID or name.

"An array of terms to set for the post, or a string of terms separated by commas. Hierarchical taxonomies must always pass IDs rather than names so that children with the same names but different parents aren’t confused."

Reply:
I don't know how else to fix the rest API error or the PHP error described in the description of the PR (Under "How?")

Reply:
If I only
1) Check the length of the postFormat attribute
2) Convert that to a comma separated string
3) Add that to the query

I get:

1) The spinner in the preview
2):

`{"code":"rest_invalid_param","message":"Invalid parameter(s): post_format","data":{"status":400,"params":{"post_format":"post_format is not a valid Term ID List. Reason: post_format[0] is not of type integer."},"details":{"post_format":{"code":"rest_no_matching_schema","message":"post_format is not a valid Term ID List. Reason: post_format[0] is not of type integer.","data":{"position":0}}}}}
`

Reply:
Thought issue indeed., I think for the whole PR we need to answer this question: Should we use post_format as a taxonomy or not? For me and given how the "post" endpoint returns the `format` property for each post and allows editing the `format` as well by passing a string.  cc @TimothyBJacobs for opinion as well. 

This means to me that ideally the `/posts` endpoint would allow passing `format` as a string in the query rather than a list of ids. 

Now, looking at this PR, it seems that we already support `post_format` in the query? and it needs to be integers. Is that correct? When was this implemented (I don't see it on the REST API handbook)

Reply:
Is the `post_format` in the query supported because you added `show_in_rest` to the taxonomy?

Reply:
I don't fully understand how it works, but yes. I am not even entirely sure which end point is being used and where I can see the code of it!

I tried disabling show_in_rest.
I also tried using 'format' instead of 'post_format'.
It did not work, with id or string.

Reply:
Ok, here's my current understanding:

 - `show_in_rest` enables access to the taxonomy in the taxonomy endpoint (using `getEntityRecords` with the `post_format` taxonomy).
 - `show_in_rest` also enables filtering the `/posts` endpoint using the taxonomy filter (using `getEntityRecords` with `post` postType and passing `post_format` in the query)

The problem for me is that this means we'll be treating the post format as a regular taxonomy but it's not the case 100% since the post object returns a `format` property in the post object.

I think the idea for me is:

 - Keep `show_in_rest` disabled for post formats.
 - Add support for a new `format` query key to the `posts` endpoint to match how the "posts" already represent the format. This likely means we need to override the Post endpoint controller here to add this and use this property in the block to filter per format. (should accept a string or a list of strings, just like the "format" property on the post object)

I would love some confirmation from folks familiar with these endpoints though. I don't want to suggest something that end up being the wrong thing :)

Reply:
I read a bit about the post formats and this last suggestion from Riad not to treat formats as a taxonomy seems a good one.

Reply:
I don't understand how to solve this.

What is "a new format query key"?  Do you mean it needs to be in get_collection_params?
https://github.com/WordPress/wordpress-develop/blob/6.6/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L2814

>Override the Post Endpoint Controller.

Does this mean something else than creating a new class, extending WP_REST_Posts_Controller?

Reply:
>What is "a new format query key"? Do you mean it needs to be in get_collection_params?

Yes, among some other changes that are going to be needed with an example to update the `get_items` function to check for the new query arg and take it into account to return the proper results.

Reply:
I have tried, but it is beyond my abilities.

Reply:
I have got something working, but I am expecting it to not pass code review 😅

Reply:
"got something working"  -except now the query has no results when no formats are used 👎